Small UI improvements

Ich73 opened the pull request:
improved positioning of elements in DenMap, EncounterLookup, IVCalculator, SeedCalculator and MainWindow,
feel free to reject all changes you don't like :)

Admiral-Fish commented:
Can you speak a little to what changes you made on each form so I know for sure?

Ich73 commented:
Sure!

**MainWindow** (MainWindow.ui): I decreased the size of the profile information group box and changed the ordering of the labels so you can better see which number is the TID, which is the SID and so on. Not a big deal but in my opinion it looks way better.

**DenMap** (DenMap.ui): I moved the combo box from the left side of the map to the bottom of the map so the whole frame is smaller. This makes it easier to leave the map open while searching a seed or something like that.

**EncounterLookup** (EncounterLookup.ui, EncounterLookup.cpp, EncounterLookup.hpp): I changed the width of the combo boxes, labels and the button so it just looks nicer. Then I added functionality to scale the first column (the locations) automatically depending on the width of the text. I had the problem not being able to read the whole location name. And the other rows get the remaining space and the text is now centered.

**IVCalculator** (IVCalculator.ui): I just moved the boxes a bit so the "Find IVs" button is now on the same row with the other inputs. If you select a pokemon with altforms the "altform" box appears directly beside the pokemon box.

**SeedCalculator** (SeedCalculator.ui, SeedCalculator.cpp, RaidFinder_**.ts): Ironically I implemented a progress bar yesterday as well :D So now I just moved the bar into the results group box and added a label below to show the search time as well as an approximation of how long it will take to complete the search. And I moved the "Use day 6" checkbox underneath the "stop after first result" checkbox so that the buttons and combo boxes are not that huge anymore.

I hope you like it.

Admiral-Fish commented on Core/Searcher/SeedSearcher.hpp:
```diff
@@ -24,6 +24,7 @@
 #include <Core/Searcher/Matrix.hpp>
 #include <QThreadPool>
 #include <QVector>
+#include <QObject>
```
Unnecessary include

Ich73 replied:
done 03215528bbd0dc15ccaa4b9d5e955c66568e1e9f

Admiral-Fish commented on Forms/Tools/EncounterLookup.hpp:
```diff
@@ -44,6 +45,7 @@ class EncounterLookup : public QWidget
     QVector<u16> speciesLookup;
 
     void setupModels();
+    void addRow(int, bool, Raid);
```
Could you add the variable names here as well

Ich73 replied:
done adc1380f456a60bae0e8a5201e6ecf3bab7bac2b

Admiral-Fish commented on Forms/Tools/SeedCalculator.cpp:
```diff
@@ -113,7 +123,20 @@ void SeedCalculator::search35()
     auto *timer = new QTimer();
-    connect(timer, &QTimer::timeout, [=] { ui->progressBar->setValue(searcher->getProgress()); });
+    auto startTime = time(0);
+    connect(timer, &QTimer::timeout, [=] {
```
Does this make the search any slower?

Ich73 replied:
"Any slower", yes, since something has to be calculated. But it is realy nothing compared to the calculation of the seeds.

Lets say that around 1000 seeds are checked every second (at least on my PC). Checking a single seed requires running the searchSeed method which has around 250 lines of code. That means that 250,000 lines of code are executed every second. The timer that updates the exstimated time is only activated every second. So there are additionally 10 lines of code every second.

That means that for a calculation that used to take 1 hour there are now 10 * 60 * 60 = 36,000 additional lines of code. Given that we execute 250,000 lines each second these extra calculations wouldn't even take an extra second.

So I would put up with that to know how long the calculation is going to take.

Admiral-Fish commented on Forms/Tools/SeedCalculator.cpp:
```diff
@@ -113,7 +123,20 @@ void SeedCalculator::search35()
+        ui->progressBar->setValue(progress);
+        auto elapsedTime = time(0) - startTime;
+        auto estimatedTime = elapsedTime * (searcher->getMaxProgress() - progress) / progress;
+        ui->progressLabel->setText(tr("elapsed time: %1:%2:%3 - estimated time: %4:%5:%6")
```
Would prefer the 'e' in elapsed and estimated to be capitalized

Ich73 replied:
done bf3a7cbef904324c8530bcfb10da652e9cf58acc
